### PR TITLE
My MathGame project

### PR DESCRIPTION
**Pull request by MartySenpai**

### CODE REVIEW

File: mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                Console.WriteLine($"Game over, Your final score is {score}. Press any key to go back to the main menu.");
+                Console.ReadLine();
+            }
+        }
+
+        timer.Start();
+        TimeSpan timeTaken = timer.Elapsed;
+
+        Helpers.AddToHistory(score, nQuestions, GameType.Addition, difficulty, timeTaken);
+    }
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs:50]: Please add one empty line between methods

File: mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                Console.WriteLine($"Game over, Your final score is {score}. Press any key to go back to the main menu.");
+                Console.ReadLine();
+            }
+        }
+        timer.Stop();
+        TimeSpan timeTaken = timer.Elapsed;
+
+        Helpers.AddToHistory(score, nQuestions, GameType.Random, difficulty, timeTaken);
+    }
+    //DivisionGame does not offer difficult levels.
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs:191]: Refrain from comments unless absolutely necessary.

File: mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+        Stopwatch timer = new Stopwatch();
+        timer.Start();
+
+        for (int i = 0; i < nQuestions; i++)
+        {
+            Console.Clear();
+            Console.WriteLine(message);
+
+            // Clear previous values from numbers.
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs:161]: Code should be self-explanatory. The less comments, the more self-explanatory your code will have to be. In this case the code explains clearly what's going on, so the comment is unnecessary. :-)

File: mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+        string difficulty = Menu.ShowDifficultyMenu();
+
+        int nQuestions = Helpers.GetAmountOfQuestions();
+
+        int score = 0;
+
+        List<char> operands = new List<char> { '+', '-', '*', '/'};
+
+        List<int> numbers = new List<int>();
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs:151]: This is a matter of personal style, but you could use something more succinct such as:
List<int> numbers = new();
var numbers = new List<int>();

File: mathGame.MartySenpai/mathGame.MartySenpai/Helpers.cs [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        Console.WriteLine("Games History");
+        Console.WriteLine("------------------------------------");
+        foreach (Game game in games)
+        {
+            Console.WriteLine($"{game.Date} - {game.Difficulty} - {game.Type} : {game.Score}/{game.NQuestions} pts - Time: {game.GameTime:mm\\:ss\\:ff}");
+        }
+        Console.WriteLine("------------------------------------\n");
+        Console.WriteLine("Press any key to return to the main menu");
+        Console.ReadLine();
+    }
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/Helpers.cs:20]: Add empty line between methods

File: mathGame.MartySenpai/mathGame.MartySenpai/Menu.cs [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                    Console.WriteLine("Goodbye!");
+                    isGameOn = false;
+                    break;
+                default:
+                    Console.WriteLine("Invalid input, press enter to continue and enter a valid key.");
+                    Console.ReadLine();
+                    break;
+            }
+        } while (isGameOn);
+    }
```
Comment (TheCSharpAcademy) [inline mathGame.MartySenpai/mathGame.MartySenpai/Menu.cs:63]: Add empty line between methods

**Review by TheCSharpAcademy — CHANGES_REQUESTED** [review]
Good work! All changes requested are in regards to code tidiness. Pay special attention to the habit of commenting on code. It's better to make your code a bit more verbose (such as when naming methods/properties/classes), than having to compensate with comments.

File: mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs (hunk 1 above)
Comment (MartySenpai) [reply mathGame.MartySenpai/mathGame.MartySenpai/GameEngine.cs:50]: Thanks you for taking your time to review, I will keep everything you have said in mind and tidy it up a bit.

I am grateful for the feedback!

**Pull request merged**